x/payment/keeper/gas_test.go

```diff
@@ -11,17 +11,38 @@ import (
 )
 
 func TestPayForDataGas(t *testing.T) {
-	app := simapp.Setup(t, false)
-	ctx := app.BaseApp.NewContext(false, tmproto.Header{})
-	k := Keeper{}
-
-	messageSize := uint64(100)
+	type testCase struct {
+		name            string
+		msg             types.MsgPayForData
+		wantGasConsumed uint64
+	}
 
-	msg := types.MsgPayForData{
-		MessageSize: messageSize,
+	testCases := []testCase{
+		{
+			name:            "1 byte message", // occupies 1 share
+			msg:             types.MsgPayForData{MessageSize: 1},
+			wantGasConsumed: uint64(4096), // 1 share * 512 bytes per share * 8 gas per byte = 4096 gas
+		},
+		{
+			name:            "100 byte message", // occupies 1 share
+			msg:             types.MsgPayForData{MessageSize: 100},
+			wantGasConsumed: uint64(4096),
+		},
+		{
+			name:            "1024 byte message", // occupies 3 shares because share prefix (e.g. namespace, info byte)
+			msg:             types.MsgPayForData{MessageSize: 1024},
+			wantGasConsumed: uint64(12288), // 3 shares * 512 bytes per share * 8 gas per byte = 12288 gas
+		},
 	}
 
-	_, err := k.PayForData(sdk.WrapSDKContext(ctx), &msg)
-	require.NoError(t, err)
-	require.Equal(t, messageSize, ctx.GasMeter().GasConsumed())
+	for _, tc := range testCases {
+		app := simapp.Setup(t, false)
```

Comment:
[non-blocking question/nit]
did the simapp have to be moved here? its probably negligible, but spinning up a new app each time seems a bit heavy.

Reply:
this line doesn't have to occur inside the `for` loop. I moved it inside for greater test isolation between test cases but it can be moved outside. I sampled test run time with this line inside / outside the for loop:

1. Inside for loop
```
ok  	github.com/celestiaorg/celestia-app/x/payment/keeper	0.430s
ok  	github.com/celestiaorg/celestia-app/x/payment/keeper	0.429s
```
2. Outside for loop
```
ok  	github.com/celestiaorg/celestia-app/x/payment/keeper	0.400s
ok  	github.com/celestiaorg/celestia-app/x/payment/keeper	0.415s
```

So it looks like tests run _slightly_ faster with the line outside the for loop. Will move outside

Reply:
resolved by https://github.com/celestiaorg/celestia-app/pull/946/commits/486a15a5fb3d17d1e68be2a0d604e6e46f577030